onnxruntime/core/framework/session_state.h

```diff
@@ -294,6 +300,10 @@ class SessionState {
     return parent_;
   }
 
+  mutable OrtMutex graph_runs_lock_;
+  mutable std::map<int64_t, std::unique_ptr<ExecutionFrame>> graph_runs_;
```

Comment:
I don't follow but I will sync offline on this to get clarity.

---
In reply to: [598976109](https://github.com/microsoft/onnxruntime/pull/7077#discussion_r598976109) [](ancestors = 598976109)

Reply:
Good catch. I have created `PartialGraphExecutionManager` to centralize all atomic operation but we still need to have a shared_ptr here to avoid holding mutex until execution is over.